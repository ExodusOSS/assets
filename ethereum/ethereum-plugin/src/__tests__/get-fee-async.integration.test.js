import { getTestingSeed, walletTester } from '@exodus/assets-testing'
import { convertForSnapshot } from '@exodus/ethereum-api/src/__tests__/eth-test-utils.js'
import { WalletAccount } from '@exodus/models'

import assetPlugin from '../index.js'

const walletAccount = WalletAccount.DEFAULT

describe(`ethereum get async fees`, () => {
  walletTester({
    assetPlugin,
    availableAssetNames: ['bat', 'ethereum', 'polygon', 'tetherusd'], // this allow reducing testing to just some accets. If not the list is massive
    assetName: 'ethereum',
    seed: getTestingSeed(),
    tests: {
      'ethereum without arguments': async ({ asset, fees: feesModule }) => {
        const fees = await feesModule.getFees({ assetName: asset.name, walletAccount })

        asset.server.stop()

        expect(convertForSnapshot(fees)).toMatchSnapshot()
      },

      'ethereum with arguments': async ({ asset, fees: feesModule }) => {
        const amount = asset.currency.parse('123 Gwei')
        const fromAddress = '0x111fffffffffffffffffffffffffffffffffffff'
        const toAddress = '0x111fffffffffffffffffffffffffffffffffffff'
        const txInput = '0x'
        const fees = await feesModule.getFees({
          assetName: asset.name,
          walletAccount,
          amount,
          fromAddress,
          toAddress,
          txInput,
        })

        asset.server.stop()

        expect(convertForSnapshot(fees)).toMatchSnapshot()
      },

      'polygon without arguments': async ({ asset, assetsModule, fees: feesModule }) => {
        const gasLimit = 38_159
        const fees = await feesModule.getFees({ assetName: 'polygon', walletAccount, gasLimit })

        asset.server.stop()

        expect(convertForSnapshot(fees)).toMatchSnapshot()
      },

      'polygon with arguments': async ({ asset, assetsModule, fees: feesModule }) => {
        const polygon = assetsModule.getAsset('polygon')
        const amount = polygon.currency.parse('0 base') // no amount
        const fromAddress = '0x111fffffffffffffffffffffffffffffffffffff'
        const toAddress = '0x111fffffffffffffffffffffffffffffffffffff'
        const fees = await feesModule.getFees({
          assetName: polygon.name,
          walletAccount,
          amount,
          fromAddress,
          toAddress,
        })

        asset.server.stop()
        expect(convertForSnapshot(fees)).toMatchSnapshot()
      },
    },
  })
})
