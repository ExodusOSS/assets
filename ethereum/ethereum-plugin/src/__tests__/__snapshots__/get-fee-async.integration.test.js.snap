exports[`ethereum get async fees ethereum with arguments 1`] = `
{
  "extraFeeData": {},
  "fee": "0.0010605 ETH",
  "gasLimit": 21000,
  "gasPrice": "0.0000000505 ETH",
  "tipGasPrice": "0.0000000005 ETH",
  "unsignedTx": {
    "txData": {
      "chainId": 1,
      "transactionBuffer": "02f00180841dcd6500850bc208d90082520894111fffffffffffffffffffffffffffffffffffff851ca35f0e0080c0808080",
    },
    "txMeta": {
      "amount": "0.000000123 ETH",
      "assetName": "ethereum",
      "bumpTxId": undefined,
      "eip1559Enabled": true,
      "fee": "0.0010605 ETH",
      "fromAddress": "0x111fffffffffffffffffffffffffffffffffffff",
      "toAddress": "0x111fffffffffffffffffffffffffffffffffffff",
    },
  },
}
`;

exports[`ethereum get async fees ethereum without arguments 1`] = `
{
  "extraFeeData": {},
  "fee": "0.0010605 ETH",
  "gasLimit": 21000,
  "gasPrice": "0.0000000505 ETH",
  "tipGasPrice": "0.0000000005 ETH",
  "unsignedTx": {
    "txData": {
      "chainId": 1,
      "transactionBuffer": "02eb0180841dcd6500850bc208d90082520894fffffffffffffffffeffffffffffffffffffffff8080c0808080",
    },
    "txMeta": {
      "amount": "0 ETH",
      "assetName": "ethereum",
      "bumpTxId": undefined,
      "eip1559Enabled": true,
      "fee": "0.0010605 ETH",
      "fromAddress": "0xF3d46F0De925B28fDa1219BbD60F5ae2a0128F9F",
      "toAddress": "0xfffffffffffffffffeffffffffffffffffffffff",
    },
  },
}
`;

exports[`ethereum get async fees polygon with arguments 1`] = `
{
  "extraFeeData": {},
  "fee": "0.0019270295 ETH",
  "gasLimit": 38159,
  "gasPrice": "0.0000000505 ETH",
  "tipGasPrice": "0.0000000005 ETH",
  "unsignedTx": {
    "txData": {
      "chainId": 1,
      "transactionBuffer": "02f8700180841dcd6500850bc208d90082950f947d1afa7b718fb893db30a3abc0cfc608aacfebb080b844a9059cbb000000000000000000000000111fffffffffffffffffffffffffffffffffffff0000000000000000000000000000000000000000000000000000000000000000c0808080",
    },
    "txMeta": {
      "amount": "0 MATIC",
      "assetName": "polygon",
      "bumpTxId": undefined,
      "eip1559Enabled": true,
      "fee": "0.0019270295 ETH",
      "fromAddress": "0x111fffffffffffffffffffffffffffffffffffff",
      "toAddress": "0x111fffffffffffffffffffffffffffffffffffff",
    },
  },
}
`;

exports[`ethereum get async fees polygon without arguments 1`] = `
{
  "extraFeeData": {},
  "fee": "0.0019270295 ETH",
  "gasLimit": 38159,
  "gasPrice": "0.0000000505 ETH",
  "tipGasPrice": "0.0000000005 ETH",
  "unsignedTx": {
    "txData": {
      "chainId": 1,
      "transactionBuffer": "02f8700180841dcd6500850bc208d90082950f947d1afa7b718fb893db30a3abc0cfc608aacfebb080b844a9059cbb000000000000000000000000fffffffffffffffffeffffffffffffffffffffff0000000000000000000000000000000000000000000000000000000000000000c0808080",
    },
    "txMeta": {
      "amount": "0 MATIC",
      "assetName": "polygon",
      "bumpTxId": undefined,
      "eip1559Enabled": true,
      "fee": "0.0019270295 ETH",
      "fromAddress": "0xF3d46F0De925B28fDa1219BbD60F5ae2a0128F9F",
      "toAddress": "0xfffffffffffffffffeffffffffffffffffffffff",
    },
  },
}
`;
